src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs

```diff
@@ -152,26 +152,29 @@ public void Start()
         {
             lock (this.lockObj)
             {
-                // Remove all transient transactions from the wallet to be re-added according to the
-                // information carried in the store. This ensures that we will re-sync in the case
-                // where the store may have been deleted.
-                // Any partial transfers affected by these removals are expected to first become
-                // suspended due to the missing wallet transactions which will rewind the counter-
-                // chain tip to then reprocess them.
-                if (this.federationWalletManager.RemoveTransientTransactions())
-                    this.federationWalletManager.SaveWallet();
-
-                Guard.Assert(this.Synchronize());
-
-                // Any transactions seen in blocks must also be present in the wallet.
-                FederationWallet wallet = this.federationWalletManager.GetWallet();
-                ICrossChainTransfer[] transfers = this.GetTransfersByStatusInternalLocked(new[] { CrossChainTransferStatus.SeenInBlock }, true, false).ToArray();
-                foreach (ICrossChainTransfer transfer in transfers)
+                lock (((FederationWalletManager)this.federationWalletManager).lockObject)
```

Comment:
I have two questions here:

1 Why is there a lock around `Start`, surely it will only get called once?
2. With this change there are now 2 locks around `Start`? This doesn't feel or seem right.

Reply:
It won't do any harm. See above.